### PR TITLE
Refactor skp_generator_tests

### PR DESCRIPTION
**Pull request by sealesj**
Refactor skp_generator_tests suite in order to reduce testing logic in test.dart and allow for later implementing package:test onto the existing skp_generator_tests tests

Part of https://github.com/flutter/flutter/issues/145482

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: dev/bots/suite_runners/run_skp_generator_tests.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../run_command.dart';
+import '../utils.dart';
+
+
```
Comment (goderbauer) [inline dev/bots/suite_runners/run_skp_generator_tests.dart:12]: nit: remove one blank line?

**Review by goderbauer — APPROVED** [review]
LGTM

File: dev/bots/suite_runners/run_skp_generator_tests.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      'core.longPaths=true',
+      'clone',
+      'https://github.com/flutter/tests.git',
+      '.',
+    ],
+    workingDirectory: checkout.path,
+  );
+  await runCommand(
+    './build.sh',
+    <String>[ ],
```
Comment (christopherfujino) [inline dev/bots/suite_runners/run_skp_generator_tests.dart:33]: uber-nit, can this be `const <String>[],`

File: dev/bots/test.dart [hunk 3]
```diff
@@ -248,7 +249,7 @@ Future<void> main(List<String> args) async {
       // All web integration tests
       'web_long_running_tests': () => webLongRunningTestsRunner(flutterRoot),
       'flutter_plugins': _runFlutterPackagesTests,
-      'skp_generator': _runSkpGeneratorTests,
+      'skp_generator': () => skpGeneratorTestsRunner(),
```
Comment (christopherfujino) [inline dev/bots/test.dart:252]: nit, you don't need the anonymous closure here, you could just tear off:

`      'skp_generator': skpGeneratorTestsRunner,`

**Review by christopherfujino — APPROVED** [review]
2 nits, but lgtm

File: dev/bots/test.dart (hunk 3 above)
Comment (sealesj) [reply dev/bots/test.dart]: Updated! Thank you

File: dev/bots/suite_runners/run_skp_generator_tests.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      'core.longPaths=true',
+      'clone',
+      'https://github.com/flutter/tests.git',
+      '.',
+    ],
+    workingDirectory: checkout.path,
+  );
+  await runCommand(
+    './build.sh',
+    const <String>[ ],
```
Comment (christopherfujino) [inline dev/bots/suite_runners/run_skp_generator_tests.dart:33]: nit `[]` (no spaces between)

Comment (sealesj) [reply dev/bots/suite_runners/run_skp_generator_tests.dart]: Whoops, fixed

**Pull request merged**